### PR TITLE
Update howto for V2 auth.

### PR DESCRIPTION
**Pull request by dbkr**

### CODE REVIEW

File: supporting-docs/howtos/client-server.rst [hunk 1]
```diff
@@ -32,31 +32,81 @@ If you already have an account, you must **login** into it.
 Registration
 ------------
 The aim of registration is to get a user ID and access token which you will need
-when accessing other APIs::
+when accessing other APIs. The Matrix specification gives a reuseable way of
```
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: Is 'reuseable' (actually spelt reusable) the right word here? Unsure what is meant.

File: supporting-docs/howtos/client-server.rst [hunk 2]
```diff
@@ -32,31 +32,81 @@ If you already have an account, you must **login** into it.
...
+        "params": {},
+        "session": "rLxMJHhGSUyWRNJEMvkLMjab"
+    }
+
+In this example, our Home Server supports two ways of authenticating. Both of
+those involve performing a single step. In one case it's ``m.login.dummy`` and in
+the other, ``m.login.email.identity``.
+
+If we wanted to allow the user to sign up with an email address, we could choose
+``m.login.email.identity``, but we're just going to choose ``m.login.dummy``. As
```
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: Do servers actually provide m.login.dummy by default? If not, the howto is misleading in that we should show people how to log in for the common default case...

File: supporting-docs/howtos/client-server.rst (hunk 1 above)
Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: I was going for the fact that the authentication part is common over all APIs that do user-interactive auth. Not sure how best to describe that.

File: supporting-docs/howtos/client-server.rst [hunk 3]
```diff
@@ -32,31 +32,81 @@ If you already have an account, you must **login** into it.
...
+    {
+        "access_token": "QGV4YW1wbGU6bG9jYWxob3N0.AqdSzFmFYrLrTmteXc",
+        "home_server": "localhost",
         "user_id": "@example:localhost"
     }
 
-NB: If a ``user`` is not specified, one will be randomly generated for you. 
-If you do not specify a ``password``, you will be unable to login to the account
-if you forget the ``access_token``.
+It worked! Note that the only reason we needed to specify dummy auth here was
```
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: what was the dummy auth that was specified here? surely the user & password params are the details for the account we're trying to register?

File: supporting-docs/howtos/client-server.rst (hunk 2 above)
Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: Yes, modulo the fact that registration will be disabled completely by default.

File: supporting-docs/howtos/client-server.rst (hunk 3 above)
Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: The dummy auth is just saying "auth": {"type": "m.login.dummy"}

**Comment by ara4n** [comment]
lgtm other than provisos inline

File: supporting-docs/howtos/client-server.rst (hunk 2 above)
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: Then we need to make it clearer from the outset that by default HSes have reg disabled and so will return nothing.  However, if they had it on and supported email-based registration and dummy registration (where nothing is validated at all...)

File: supporting-docs/howtos/client-server.rst (hunk 3 above)
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: ok.... then i don't understand the explanation that "we can't call an API that uses user-interactive authentication without supplying some kind of authentication".  OH! you mean that the reason to register for an account is to use APIs which need you to be logged in?  Rewording could be a good thing here if so...

File: supporting-docs/howtos/client-server.rst (hunk 1 above)
Comment (ara4n) [inline supporting-docs/howtos/client-server.rst]: "Matrix specification gives an extensible way of allowing users to present authentication for API calls"?

Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: wfm: changed

File: supporting-docs/howtos/client-server.rst (hunk 2 above)
Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: Actually servers will still offer dummy auth even with registration disabled: it just won't work, as per the bit I added below.

File: supporting-docs/howtos/client-server.rst (hunk 3 above)
Comment (dbkr) [inline supporting-docs/howtos/client-server.rst]: Nope, I just mean that in order to call the API and it do thing thing you want to happen, you need an 'auth' key in your dictionary with something in it the server has told you it supports.

File: supporting-docs/howtos/client-server.rst [hunk 4]
```diff
@@ -73,15 +123,15 @@ The aim when logging in is to get an access token for your existing user ID::
...
-login. The reference home server implementation support username/password login,
-but other home servers may support different login methods (e.g. OAuth2).
 
+Notice that we still start by asking the server what types of auth we can use,
+then perform the call. In V1 the initial query happens with a GET request and
+our authentication information is not in its own 'auth' subdictionary. There are
+some other differences between the V1 and V2 login spec too which you can read
+about in the full spec documentation.
+
+.. TODO: Mention fallback auth once this is all V2.
```
Comment (Kegsay) [inline supporting-docs/howtos/client-server.rst]: Does this mean fallback is incompatible with v2_alpha?

**Comment by dbkr** [comment]
Are questions here still valid? Should this be merged?

**Pull request closed without merging**